.github/actions/install-wdk/action.yml

```diff
@@ -130,16 +130,154 @@ runs:
           Write-Output "WindowsSdkBinPath=$packages\Microsoft.Windows.SDK.CPP.$version\c\bin" >> $env:GITHUB_ENV
         } else {
 
-          Write-Host "Using Winget to install WDK version $inputVersion..."
-          $wdk = "Microsoft.WindowsWDK.$inputVersion"
-          if ((Get-WinGetPackage -Id $wdk -Source winget -MatchOption Equals).Id -eq '$wdk') {
-            Write-Host "$wdk is already installed. Attempting to update..."
-            Update-WinGetPackage -Id $wdk -Source winget -MatchOption Equals -Mode Silent -Force
+        Write-Host "Using Winget to install WDK version $inputVersion..."
+        
+        # Import the WinGet PowerShell module
+        Write-Host "Importing WinGet PowerShell module..."
+        Import-Module Microsoft.WinGet.Client -Force
+        
+        # Detect and uninstall any existing Windows SDK packages to avoid conflicts
+        Write-Host "Detecting and removing existing Windows SDK packages..."
+        try {
+          $existingSDKs = Get-WinGetPackage | Where-Object { $_.Name -like "*Windows Software Development Kit*" -or $_.Name -like "*Windows SDK*" -or $_.Id -like "Microsoft.WindowsSDK*" }
+          if ($existingSDKs.Count -gt 0) {
+            Write-Host "Detected $($existingSDKs.Count) existing Windows SDK packages:"
+            foreach ($sdk in $existingSDKs) {
+              Write-Host "  - Name: $($sdk.Name), ID: $($sdk.Id), Version: $($sdk.InstalledVersion)"
+            }
+          } else {
+            Write-Host "No existing Windows SDK packages found."
+          }
+        } catch {
+          Write-Error "Failed to query for existing Windows SDK packages: ${_}"
+          Write-Host "Try manually running: Get-WinGetPackage | Where-Object { `$_.Name -like '*Windows SDK*' -or `$_.Id -like 'Microsoft.WindowsSDK*' }"
+          exit 1
+        }
+        foreach ($sdk in $existingSDKs) {
+          Write-Host "Uninstalling existing Windows SDK: Name=$($sdk.Name), ID=$($sdk.Id), Version=$($sdk.InstalledVersion)"
+          try {
+            $uninstallResult = Uninstall-WinGetPackage $sdk.Id -Mode Silent -Force
+            Write-Host "Uninstall output: $uninstallResult"
```

Comment:
this ends up looking like `Uninstall output: Microsoft.WinGet.Client.Engine.PSObjects.PSUninstallResult`. thats not very helpful..